### PR TITLE
SALTO-2895 SALTO-2894 support fetching custom statuses

### PR DESCRIPTION
**Pull request by netama**
When custom statuses are enabled in the account, a `Ticket status` ticket field is added with a `custom_statuses` field that causes the workspace to break.
This is the first step - fetching ticket statuses explicitly, and omitting them from the `Ticket status` field. We will follow up with deploy support + add back references from the ticket field.

---
_Release Notes_: 
_Zendesk adapter:_
* Fix bug where fetching from accounts with custom statuses enabled would result in workspace errors.

---
_User Notifications_: 
_Zendesk adapter:_
* Instances of `Custom status` will be added in the workspace for accounts where the feature is enabled

### CODE REVIEW

**Review by edenhassid — APPROVED** [review]

File: packages/zendesk-adapter/src/filters/dynamic_content_references.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ const transformDynamicContentDependencies = async (
       if (!enableMissingReference) {
         return [part]
       }
-      const matches = placeholder[0].match(/.*\.([a-zA-Z0-9_]+)\}\}/)
+      const matches = placeholder[0].match(/dc\.([a-zA-Z0-9_-]+)\}\}/)
```
Comment (netama) [inline packages/zendesk-adapter/src/filters/dynamic_content_references.ts:60]: unrelated fix - seems this caught too much
